pkg/autohealing/cloudprovider/openstack/provider.go

```diff
@@ -288,15 +289,40 @@ func (provider CloudProvider) firstTimeRepair(n healthcheck.NodeInfo, serverID s
 			// Uncordon the node
 			if n.IsWorker {
 				nodeName := n.KubeNode.Name
-				newNode := n.KubeNode.DeepCopy()
-				newNode.Spec.Unschedulable = false
-				if _, err := provider.KubeClient.CoreV1().Nodes().Update(context.TODO(), newNode, metav1.UpdateOptions{}); err != nil {
-					log.Errorf("Failed to cordon node %s, error: %v", nodeName, err)
-				} else {
-					log.Infof("Node %s is cordoned", nodeName)
+				// timeout for wait.Poll
+				ctx := context.Background()
+				errServiceUp := wait.PollUntilContextTimeout(ctx, 3*time.Second, provider.Config.RebuildDelayAfterReboot, false,
+					func(ctx context.Context) (bool, error) {
+						rapairedNode, getErr := provider.KubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+						if getErr != nil {
+							log.Errorf("Failed to get node %s, error: %v in wait.PollUntilContextTimeout waiting for services to start", nodeName, getErr)
+							return false, getErr
+						}
+						if strings.Contains(rapairedNode.Status.String(), "Type:Ready,Status:True") {
```

Comment:
Wait, we're dumping an object into a string to check some property? This is pretty awful, let's do it properly, by comparing the object properties. You can do it in a helper function to make this code clear.

Reply:
Created a helper function to check the node condition.